test/conformance/ingress/update_test.go

```diff
@@ -192,8 +197,11 @@ func checkOK(t *testing.T, url string, client *http.Client) context.CancelFunc {
 	}()
 
 	// Return a cancel function that stops the prober and then waits for it to complete.
+	var stopOnce sync.Once
 	return func() {
-		close(stopCh)
-		<-doneCh
+		stopOnce.Do(func() {
+			close(stopCh)
+			<-doneCh
+		})
```

Comment:
I got the impression from the previous PR that the problem was this not getting called, not that it was called more than once.  cc @k4leung4 @vagababov 

Personally, I'd just rollback the other PR until we have more info on the panic to roll forward.

Reply:
With the changes in this PR, it can be called twice because the `defer` here: https://github.com/knative/serving/pull/7470/files#diff-e4259a59a6b444de6cfaff00bbf2cfb5L70

By reading the description of the previous PR, I got the impression that the `defer` was needed. If it's not, we can drop this one and revert the other.

Reply:
I forgot the context, but this looks strange. We call  `probeCancel` twice. But instead we should probanly just do the
```
defer func() {
  proberCancel()
  previousVersionCancel()
}()
```

Reply:
I've changed this to use a single defer. Check my latest commit.